omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -171,17 +176,21 @@ private SimpleObject parseDisposition(DispositionDescriptor dispositionDescripto
 		return simpleObject;
 	}
 	
-	private SimpleObject parseDiagnosis(DiagnosisMetadata diagnosisMetadata, Obs obs) {
-		Diagnosis diagnosis = diagnosisMetadata.toDiagnosis(obs);
+	private SimpleObject parseDiagnosis(Encounter encounter) {
+		
+		List<Diagnosis> diag = diagnosisServiceCompatibility.getPrimaryDiagnosis(encounter);
 		
-		String answer = "(" + uiUtils.message("coreapps.Diagnosis.Certainty." + diagnosis.getCertainty()) + ") ";
-		answer += diagnosis.getDiagnosis().formatWithCode(uiUtils.getLocale(),
-		    emrApiProperties.getConceptSourcesForDiagnosisSearch());
+		SimpleObject simpleObject = SimpleObject.fromObject(encounter, uiUtils, "encounterId");
```

Comment:
hi @ibacher @dkayiwa ,this whole line is returning a null, could i be missing out some thing?